src/drt/src/io/io_parser_helper.cpp

```diff
@@ -45,6 +45,29 @@ using namespace boost::polygon::operators;
 using Rectangle = boost::polygon::rectangle_data<int>;
 namespace gtl = boost::polygon;
 
+bool io::Parser::checkPinsAboveTopRoutingLayer()
+{
+  if (design_->getTopBlock()) {
+    for (const auto& bTerm : design_->getTopBlock()->getTerms()) {
+      if (bTerm->isAboveTopLayer()) {
+        return true;
+      }
+    }
+    for (const auto& inst : design_->getTopBlock()->getInsts()) {
+      for (const auto& iTerm : inst->getInstTerms()) {
+        for (const auto& pin : iTerm->getTerm()->getPins()) {
+          for (const auto& fig : pin->getFigs()) {
+            if (((frShape*) (fig.get()))->getLayerNum() > TOP_ROUTING_LAYER) {
+              return true;
+            }
+          }
+        }
+      }
+    }
+  }
```

Comment:
Add a check to see if the pin is connected to a net that is to be routed by drt. If it's not then just skip it.